src/config.py

```diff
@@ -205,18 +223,33 @@ def server_properties(self) -> List[str]:
             + DEFAULT_CONFIG_OPTIONS.split("\n")
         )
 
-    def push(self, content: str, path: str) -> None:
-        """Wrapper for writing a file and contents to a container.
+        if not self.charm.tls.enabled:
+            properties += [
+                "listeners=SASL_PLAINTEXT://:9092",
+                f"advertised.listeners=SASL_PLAINTEXT://{host}:9092",
+                f'listener.name.sasl_plaintext.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="sync" password="{self.sync_password}";',
+                "security.inter.broker.protocol=SASL_PLAINTEXT",
+                "security.protocol=SASL_PLAINTEXT",
+            ]
+        else:
+            properties += [
+                "listeners=SASL_SSL://:9093",
+                f"advertised.listeners=SASL_SSL://{host}:9093",
+                f'listener.name.sasl_ssl.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="sync" password="{self.sync_password}";',
+                "security.inter.broker.protocol=SASL_SSL",
+                "security.protocol=SASL_SSL",
+            ]
+            properties += self.tls_properties
```

Comment:
personal nit, could these look like:

```python
port = 9093 if self.charm.tls.enabled else 9092
protocol = "SASL_SSL" if self.charm.tls.enabled else "SASL_PLAINTEXT"
listener_name = lower(protocol)

properties += [
    f"listeners={protocol}://:{port}",
    f"advertised.listeners=protocol://{host}:{port}",
    f'listener.name.{protocol}.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="sync" password="{self.sync_password}";',
    f"security.inter.broker.protocol={protocol}",
    f"security.protocol={protocol}",
]
```

Reply:
hm, yeah I like that better. Will update